File: Assignment8/Program4.c
// 4.Write a program which accept range from user and return addition of all even numbers
// in between that range. (Range should contains positive numbers only)
// Input :  23 30  Output :  108
// Input :  10 18 Output :  70
// Input :  -10 2 Output :  Invalid range
// Input :  90 18 Output :  Invalid range

#include <stdio.h>

int RangeSumEven(int iStart, int iEnd)
{
    int iCnt = 0;
    int iSum = 0;

    for (iCnt = iStart; iCnt <= iEnd; iCnt++)
    {
        if (iCnt % 2 == 0)
        {
            iSum = iSum + iCnt;
        }
    }
    return iSum;
}
int main()
{
    int iValue1 = 0, iValue2 = 0, iRet = 0;

    printf("Enter starting point");
    scanf("%d", &iValue1);
    printf("Enter ending point");
    scanf("%d", &iValue2);

    iRet = RangeSumEven(iValue1, iValue2);
    
    if (iValue1 < 0 || iValue1 > iValue2)
    {
        printf("Invalid range\n");
    }
    else
    {
        printf("Addition is %d", iRet);
    }

    return 0;
}

File: Assignment32/Program4.java
// 4. Write Java program which accept N numbers from user and display all such elements
//  which are divisible by 3 and 5. 

// Input :  N :   6 
//   Elements : 85 66 3 15 93 88  
// Output : 15 

import java.util.*;

class Program4
{
	public static void main(String[] args) 
	{
		Scanner sobj = new Scanner(System.in);
		int iSize = 0,iCnt = 0;

		System.out.println("Enter Number of Elements");
		iSize = sobj.nextInt();

		int Arr[] = new int[iSize];

		System.out.println("Enter Elements");
		for(iCnt = 0;iCnt < Arr.length;iCnt++)
		{
			Arr[iCnt] = sobj.nextInt();
		}

		ArrayDemo adobj = new ArrayDemo();
		adobj.Display(Arr);
	}
}

class ArrayDemo 
{  
    public void Display(int Arr[])  
    {   
    	int iCnt = 0;

    	for(iCnt = 0;iCnt < Arr.length;iCnt++)
    	{
    		if(Arr[iCnt] % 3 == 0 && Arr[iCnt] % 5 == 0)
    		{
    			System.out.println("Elements Divisible by 3 & 5 are :"+Arr[iCnt]);
    		}
    	}
    } 
} 